commonItems/Parser.cs

```diff
@@ -0,0 +1,287 @@
+                } else if (!inQuotes && !inLiteralQuote && inputChar == '{') {
+                    if (sb.Length == 0) {
+                        sb.Append(inputChar);
+                    } else {
+                        reader.PushBack('{');
+                    }
+                    break;
+                } else if (!inQuotes && !inLiteralQuote && inputChar == '}') {
+                    if (sb.Length == 0) {
+                        sb.Append(inputChar);
+                    } else {
+                        reader.PushBack('}');
+                    }
+                    break;
+                } else if (!inQuotes && !inLiteralQuote && inputChar == '=') {
+                    if (sb.Length == 0) {
+                        sb.Append(inputChar);
+                    } else {
+                        reader.PushBack('=');
+                    }
+                    break;
+                } else {
+                    sb.Append(inputChar);
+                }
+
+                previousCharacter = inputChar;
+            }
+            return sb.ToString();
+        }
+
+        public static string? GetNextTokenWithoutMatching(BufferedReader reader) {
+            string? toReturn = null;
+            var gotToken = false;
+            while (!gotToken) {
+                if (reader.EndOfStream) {
+                    return null;
+                }
+                toReturn = GetNextLexeme(reader);
+                gotToken = true;
+            }
+
+            return toReturn;
+        }
+
+        public string? GetNextToken(BufferedReader reader) {
+            var sb = new StringBuilder();
+
+            var gotToken = false;
+            while (!gotToken) {
+                if (reader.EndOfStream) {
+                    return null;
+                }
+
+                sb.Length = 0;
+                sb.Append(GetNextLexeme(reader));
+
+                var strippedToken = RemQuotes(sb.ToString());
+                var isTokenQuoted = (strippedToken.Length < sb.ToString().Length);
+
+                var matched = TryToMatch(sb.ToString(), strippedToken, isTokenQuoted, reader);
+
+                if (!matched) {
+                    gotToken = true;
+                }
+            }
+            return sb.Length == 0 ? null : sb.ToString();
+        }
+
+        public void ParseStream(BufferedReader reader) {
+            var braceDepth = 0;
+            var value = false; // tracker to indicate whether we reached the value part of key=value pair
+            var tokensSoFar = new StringBuilder();
+
+            while (true) {
+                var token = GetNextToken(reader);
+                if (token != null) {
+                    tokensSoFar.Append(token);
+                    if (token == "=") {
+                        if (!value) {
+                            value = true; // swapping to value part.
+                            continue;
+                        } else { // leaving else to be noticeable.
```

Comment:
Ditto, though that does violate the
```
} else {
````
convention (which is one I dislike for reasons like this, but my like or dislike shouldn't necessarily force a style change)